llvm/test/CodeGen/X86/vaargs.ll

```diff
@@ -1,4 +1,5 @@
-; RUN: llc -verify-machineinstrs -mcpu=corei7-avx %s -o - | FileCheck %s --check-prefix=CHECK --check-prefix=NO-FLAGS
+; RUN: llc -verify-machineinstrs -mcpu=corei7-avx -disable-x86-lea-opt %s -o - | FileCheck %s --check-prefix=CHECK --check-prefix=NO-FLAGS
+; Disable the x86-optimize-LEAs pass, since it may introduce an intermediate LEA instruction that changes the base register of the vmovaps instructions from %rsp to something else.
```

Comment:
Why do we need the change here? I understand the patch is NFC.

Reply:
I reverted the changes in X86OptimizeLEAs.cpp (it uses X86II::getMemoryOperandNo again) and the changes to this test. It looks like X86OptimizeLEAs can't handle pseudo instructions like VASTART_SAVE_XMM_REGS that may expand to multiple memory accesses at different displacements.